apis/condition_set.go

```diff
@@ -148,6 +159,29 @@ func (r conditionsImpl) IsHappy() bool {
 	return true
 }
 
+// IsUnknown looks at the happy condition and returns true if that condition is
+// set to Unknown or that condition is nil.
+func (r conditionsImpl) IsUnknown() bool {
```

Comment:
Implementation does not match the comment. If there's a `nil` it'll panic.
Also you can `return c == nil || c.IsUnknown`
Same below.

Reply:
c.isUnknown() will check whether c is `nil`, if c is `nil`, it will return true. It won't panic.

Reply:
Do you think the comment here is confusing since it touches the implementation details of c.isUnknown()?

Reply:
yes it presumes some things about internal implementation.,

Reply:
I will modify the comment here to 
// IsUnknown looks at the happy condition and returns true if that condition is set to Unknown

Reply:
sure, what about `return r.GetCondition(r.happy).IsUnknown()` as your whole method?